src/ucp/dt/datatype_iter.inl

```diff
@@ -504,6 +504,8 @@ ucp_datatype_iter_mem_reg(ucp_context_h context, ucp_datatype_iter_t *dt_iter,
                             uct_flags, &dt_iter->type.contig.memh);
     } else if (ucp_datatype_iter_is_class(dt_iter, UCP_DATATYPE_IOV, dt_mask)) {
         return ucp_datatype_iter_iov_mem_reg(context, dt_iter, md_map, uct_flags);
+    } else if (uct_flags & UCT_MD_MEM_FLAG_HIDE_ERRORS) {
+        return UCS_OK;
```

Comment:
what test is failed? I guess this flags is intended to hide possible registration errors on the specific MDs, while using incorrect datatype looks like a SW bug which needs to be reported

Reply:
We perform memory registration inside `ucp_proto_rndv_rts_request_init`.
Previously I added check for `GENERIC` datatype, and replaced it by check for the flag according to the https://github.com/openucx/ucx/pull/8131#discussion_r851996601

Reply:
i think checking for `GENERIC` datatype was cleaner. Using `UCT_MD_MEM_FLAG_HIDE_ERRORS` for checking proper datatype looks error prone to me. We could just remove the error from this function

Reply:
when this function is called without HIDE_ERRORS it means the caller expects registration to really succeed, IMO if we just remove the error it can hide real issues

Reply:
but this `HIDE` flag is intended for other type of issues. What if we pass invalid datatype, due to a SW bug together with `HIDE` flag? Imo, hiding such errors is not a good approach

Reply:
so maybe add another wrapper inline function , so sometimes we would fail if datatype is invalid, and sometimes don't?